api/tests/controllers/recipes.test.js

```diff
@@ -1,45 +1,229 @@
+    email: "poppy@email.com",
+    password: "Password1!",
+  });
+  const recipe = new Recipe({
+    name: "Easy blueberry muffins",
+    description:
+      "Follow this easy blueberry muffins recipe for a fun weekend bake. You're just a few simple steps away from bite-sized blueberry treats",
+    ownerId: "65e1ed2534ba17d5e7b48f68",
+    tags: ["Baking", "Blueberry"],
+    favouritedByOwner: false,
+    totalTime: 40,
+    recipeYield: 4,
+    recipeIngredient: [
+      "100g unsalted butter softened, plus 1 tbsp, melted, for greasing",
+      "140g golden caster sugar",
+      "2 large eggs",
+      "140g natural yogurt",
+      "1 tsp vanilla extract",
+      "2 tbsp milk",
+      "250g plain flour",
+      "2 tsp baking powder",
+      "1 tsp bicarbonate of soda",
+      "125g pack blueberries (or use frozen)",
+    ],
+    recipeInstructions: [
+      "Heat oven to 200C/180C fan/gas 6 and line a 12-hole muffin tin with paper cases. Beat the butter and caster sugar together until pale and fluffy. Add the eggs and beat in for 1 min, then mix in the yogurt, vanilla extract and milk. Combine the flour, baking powder and bicarb in a bowl with ¼ tsp fine salt, then tip this into the wet ingredients and stir in. Finally, fold in the blueberries and divide the mixture between the muffin cases.",
+      "Bake for 5 mins, then reduce oven to 180C/160C fan/gas 4 and bake for 15-18 mins more until risen and golden, and a cocktail stick inserted into the centre comes out clean.",
+      "Cool in the tin for 10 mins, then carefully lift out onto a wire rack to finish cooling. Will keep for 3-4 days in an airtight container – after a day or two, pop them in the microwave for 10-15 secs on high to freshen up.",
+    ],
+    url: "https://www.bbcgoodfood.com/recipes/blueberry-muffin",
+    image:
+      "https://images.immediate.co.uk/production/volatile/sites/30/2020/08/blueberry-muffin-ee95fd4.jpg?resize=768,574",
+    dateAdded: dateAdded,
+  });
+
+  beforeAll(async () => {
+    await user.save();
+    await recipe.save();
+    token = createToken(user.id);
+    console.log("token ----------------------", token);
+  });
 
-    const req = { query: { url: "https://example.com" } };
-    const res = { status: jest.fn().mockReturnThis(), json: jest.fn() };
+  afterAll(async () => {
+    await User.deleteMany({});
+    await Recipe.deleteMany({});
+  });
 
-    await RecipesController.fetchRecipeData(req, res);
+  describe("GET, when token is present", () => {
+    test("the response code is 200", async () => {
+      const response = await request(app)
+        .get(`/recipes/${recipe._id}`)
+        .set("Authorization", `Bearer ${token}`);
+
+      expect(response.status).toEqual(200);
+    });
+
+    test("returns recipe data for that id", async () => {
+      const response = await request(app)
+        .get(`/recipes/${recipe._id}`)
+        .set("Authorization", `Bearer ${token}`);
+
+      const recipeData = response.body.recipeData;
+
```

Comment:
I think can refactor to expect if the recipeData from the backend = the const recipe = new Recipe. Can be done later. Otherwise thumbs up!